tools/pythonpkg/tests/fast/test_alex_multithread.py

```diff
@@ -0,0 +1,93 @@
+
+class TestCursorMultithread(object):
+    def test_alex_first(self, duckdb_cursor):
+        duckdb_con = duckdb.connect(check_same_thread=False) # In Memory DuckDB
+        duckdb_con.execute("""CREATE OR REPLACE TABLE my_inserts (thread_name varchar)""")
+
+        thread_count = 3
+        threads = []
+
+        # Kick off multiple threads (in the same process) 
+        # Pass in the same connection as an argument, and an object to store the results
+        for i in range(thread_count):
+            threads.append(Thread(target=insert_from_thread,
+                                    args=(duckdb_con,),
+                                    name='my_thread_'+str(i)))
+
+        for i in range(thread_count):
+            threads[i].start()
+
+        for i in range(thread_count):
+            threads[i].join()
+
+        assert duckdb_con.execute("""SELECT * FROM my_inserts order by thread_name""").fetchall() == [('my_thread_0',), ('my_thread_1',), ('my_thread_2',)]
+
+    def test_alex_second(self, duckdb_cursor):
+        duckdb_con = duckdb.connect(check_same_thread=False) # In Memory DuckDB
+        duckdb_con.execute("""CREATE OR REPLACE TABLE my_inserts (thread_name varchar)""")
+
+        thread_count = 3
+        threads = []
+        cursors = []
+
+        # Kick off multiple threads (in the same process) 
+        # Pass in the same connection as an argument, and an object to store the results
+        for i in range(thread_count):
+            cursors.append(duckdb_con.cursor())
+            threads.append(Thread(target=insert_from_thread_second,
+                                    args=(cursors[i],),
+                                    name='my_thread_'+str(i)))
+
+        for i in range(thread_count):
+            threads[i].start()
+
+        for i in range(thread_count):
+            threads[i].join()
+
+        assert duckdb_con.execute("""SELECT * FROM my_inserts order by thread_name""").fetchall() == [('my_thread_0',), ('my_thread_1',), ('my_thread_2',)]
+
+    def test_alex_third(self, duckdb_cursor):
+        duckdb_con = duckdb.connect('another_test_10.duckdb', check_same_thread=False)
+        duckdb_con.execute("""CREATE OR REPLACE TABLE my_inserts (thread_name varchar)""")
+
+
+        thread_count = 3
+        threads = []
+
+        # Kick off multiple threads (in the same process) 
+        # Pass in the same connection as an argument, and an object to store the results
+        for i in range(thread_count):
+            threads.append(Thread(target=insert_from_thread_third,
+                                    args=(duckdb_con,),
+                                    name='my_thread_'+str(i)))
+
+        for i in range(thread_count):
+            threads[i].start()
+
+        for i in range(thread_count):
+            threads[i].join()
+
+        assert duckdb_con.execute("""SELECT * FROM my_inserts order by thread_name""").fetchall() == [('my_thread_0',), ('my_thread_1',), ('my_thread_2',)]
```

Comment:
Use the assertion methods in the unittest library instead of the assert keyword?